src/xrCore/xrDebug.h

```diff
@@ -78,13 +74,27 @@ class XRCORE_API xrDebug
         const char* arg1 = nullptr, const char* arg2 = nullptr);
     static void DoExit(const std::string& message);
 
+	///
+	/// Note: DbgHelp is singlethreaded, so you must synchronize calls to these functions
```

Comment:
Why not wrap the content of LogStackTrace into a critical section?

Reply:
I didn't find a RAII compatible implementation in the current codebase. So for minimizing size of the changes at now, I left the sync question for the caller

Reply:
Lock isn't RAII compatible? :(

Reply:
```
struct LockImpl
{
    std::recursive_mutex mutex;
};

Lock::~Lock()
{
    delete impl;    /// impl is LockImpl
}
```

> The behavior of a program is undefined if a recursive_mutex is destroyed while still owned by some thread.